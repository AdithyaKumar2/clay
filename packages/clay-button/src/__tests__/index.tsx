/**
 * SPDX-FileCopyrightText: Â© 2019 Liferay, Inc. <https://liferay.com>
 * SPDX-License-Identifier: BSD-3-Clause
 */

import ClayButton, {ClayButtonWithIcon} from '..';
import {cleanup, render} from '@testing-library/react';
import React from 'react';

describe('ClayButton', () => {
	afterEach(cleanup);

	it('renders', () => {
		const {container} = render(<ClayButton />);

		expect(container).toMatchSnapshot();
	});

	it('renders with a different display type', () => {
		const {container} = render(<ClayButton displayType="link" />);

		expect(container).toMatchSnapshot();
	});

	it('renders disabled', () => {
		const {container} = render(
			<ClayButton disabled displayType="primary" />
		);

		expect(container).toMatchSnapshot();
	});

	it('renders borderless', () => {
		const {container} = render(
			<ClayButton borderless displayType="primary" />
		);

		expect(container).toMatchSnapshot();
	});

	it('renders block', () => {
		const {container} = render(<ClayButton block displayType="primary" />);

		expect(container).toMatchSnapshot();
	});

	it('renders outline', () => {
		const {container} = render(
			<ClayButton displayType="primary" outline />
		);

		expect(container).toMatchSnapshot();
	});

	it('renders monospaced', () => {
		const {container} = render(
			<ClayButton displayType="primary" monospaced />
		);

		expect(container).toMatchSnapshot();
	});

	it('renders small', () => {
		const {container} = render(<ClayButton displayType="primary" small />);

		expect(container).toMatchSnapshot();
	});

	it('renders with a ButtonGroup', () => {
		const {container} = render(
			<ClayButton.Group>
				<ClayButton />
			</ClayButton.Group>
		);

		expect(container).toMatchSnapshot();
	});

	it('renders a ButtonGroup with spaced', () => {
		const {container} = render(
			<ClayButton.Group spaced>
				<ClayButton />
				<ClayButton />
			</ClayButton.Group>
		);

		expect(container).toMatchSnapshot();
	});

	it('renders ButtonWithIcon', () => {
		const {container} = render(
			<ClayButtonWithIcon
				aria-label="Delete"
				spritemap="/some/path"
				symbol="trash"
			/>
		);

		expect(container).toMatchSnapshot();
	});

	it('renders ButtonWithIcon without monospaced', () => {
		const {container} = render(
			<ClayButtonWithIcon
				aria-label="Delete"
				monospaced={false}
				spritemap="/some/path"
				symbol="trash"
			/>
		);

		expect(container).toMatchSnapshot();
	});
});
