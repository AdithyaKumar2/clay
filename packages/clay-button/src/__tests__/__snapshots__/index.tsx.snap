// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ClayButton renders 1`] = `
<div>
  <button
    class="btn btn-primary"
    type="button"
  />
</div>
`;

exports[`ClayButton renders ButtonWithIcon 1`] = `
<div>
  <button
    aria-label="Delete"
    class="btn btn-monospaced btn-primary"
    type="button"
  >
    <svg
      class="lexicon-icon lexicon-icon-trash"
      role="presentation"
    >
      <use
        xlink:href="/some/path#trash"
      />
    </svg>
  </button>
</div>
`;

exports[`ClayButton renders ButtonWithIcon without monospaced 1`] = `
<div>
  <button
    aria-label="Delete"
    class="btn btn-primary"
    type="button"
  >
    <svg
      class="lexicon-icon lexicon-icon-trash"
      role="presentation"
    >
      <use
        xlink:href="/some/path#trash"
      />
    </svg>
  </button>
</div>
`;

exports[`ClayButton renders a ButtonGroup with spaced 1`] = `
<div>
  <div
    class="btn-group"
    role="group"
  >
    <div
      class="btn-group-item"
    >
      <button
        class="btn btn-primary"
        type="button"
      />
    </div>
    <div
      class="btn-group-item"
    >
      <button
        class="btn btn-primary"
        type="button"
      />
    </div>
  </div>
</div>
`;

exports[`ClayButton renders block 1`] = `
<div>
  <button
    class="btn btn-block btn-primary"
    type="button"
  />
</div>
`;

exports[`ClayButton renders borderless 1`] = `
<div>
  <button
    class="btn btn-outline-borderless btn-outline-primary"
    type="button"
  />
</div>
`;

exports[`ClayButton renders disabled 1`] = `
<div>
  <button
    class="btn btn-primary"
    disabled=""
    type="button"
  />
</div>
`;

exports[`ClayButton renders monospaced 1`] = `
<div>
  <button
    class="btn btn-monospaced btn-primary"
    type="button"
  />
</div>
`;

exports[`ClayButton renders outline 1`] = `
<div>
  <button
    class="btn btn-outline-primary"
    type="button"
  />
</div>
`;

exports[`ClayButton renders small 1`] = `
<div>
  <button
    class="btn btn-sm btn-primary"
    type="button"
  />
</div>
`;

exports[`ClayButton renders with a ButtonGroup 1`] = `
<div>
  <div
    class="btn-group"
    role="group"
  >
    <button
      class="btn btn-primary"
      type="button"
    />
  </div>
</div>
`;

exports[`ClayButton renders with a different display type 1`] = `
<div>
  <button
    class="btn btn-link"
    type="button"
  />
</div>
`;
